chia/wallet/did_wallet/did_wallet_puzzles.py

```diff
@@ -75,6 +85,21 @@ def create_fullpuz(innerpuz: Program, launcher_id: bytes32) -> Program:
     return SINGLETON_TOP_LAYER_MOD.curry(singleton_struct, innerpuz)
 
 
+def create_fullpuz_hash(innerpuz_hash: bytes32, launcher_id: bytes32) -> bytes32:
+    """
+    Create a full puzzle of DID
+    :param innerpuz_hash: DID inner puzzle tree hash
+    :param launcher_id: launcher coin name
+    :return: DID full puzzle hash
+    """
+    mod_hash = SINGLETON_TOP_LAYER_MOD.get_tree_hash()
+    quoted_mod_hash = calculate_hash_of_quoted_mod_hash(SINGLETON_TOP_LAYER_MOD.get_tree_hash())
+    # singleton_struct = (MOD_HASH . (LAUNCHER_ID . LAUNCHER_PUZZLE_HASH))
+    singleton_struct = Program.to((mod_hash, (launcher_id, LAUNCHER_PUZZLE.get_tree_hash())))
```

Comment:
it's a bit of a nit-pick, but it seems unnecessary to compute tree-hashes of constants every time we want a new puzzle hash. We do that 3 times in here.

Reply:
lets fix this in a separate PR